core/src/main/java/org/bitcoinj/core/Block.java

```diff
@@ -178,13 +179,31 @@ private static List<Transaction> readTransactions(ByteBuffer payload) throws Buf
         return transactions;
     }
 
-    /** Special case constructor, used for the genesis node, cloneAsHeader and unit tests. */
+    /** Special case constructor, used for unit tests. */
+    @VisibleForTesting
     Block(long setVersion) {
         // Set up a few basic things. We are not complete after this though.
-        version = setVersion;
-        difficultyTarget = 0x1d07fff8L;
-        time = TimeUtils.currentTime().truncatedTo(ChronoUnit.SECONDS); // convert to Bitcoin time
-        prevBlockHash = Sha256Hash.ZERO_HASH;
+        this(setVersion,
+                TimeUtils.currentTime().truncatedTo(ChronoUnit.SECONDS), // convert to Bitcoin time)
+                0x1d07fff8L,
+                0,
+                Collections.emptyList());
+    }
+
+    // For unit-test genesis blocks
```

Comment:
Should this be `@VisibleForTesting` then?

Reply:
Yes. I'll update it.